src/test/java/org/elasticsearch/test/rest/section/SkipSection.java

```diff
@@ -57,27 +74,39 @@ public String getReason() {
         return reason;
     }
 
-    public boolean skip(String currentVersion) {
+    public boolean skip(Version currentVersion) {
         if (isEmpty()) {
             return false;
         }
-
-        if (version != null) {
-            return VersionUtils.skipCurrentVersion(version, currentVersion);
-        }
-
-        if (features != null && !this.features.isEmpty()) {
-            return !Features.areAllSupported(this.features);
+        if (isVersionCheck()) {
+            return currentVersion.onOrAfter(lowerVersion) && currentVersion.onOrBefore(upperVersion);
+        } else {
+            return Features.areAllSupported(features) == false;
         }
-
-        throw new IllegalArgumentException("version or feature should be not null in a non empty skip section");
     }
 
     public boolean isVersionCheck() {
-        return Strings.hasLength(version);
+        return features.isEmpty();
     }
 
     public boolean isEmpty() {
         return EMPTY.equals(this);
     }
+    
+    private Version[] parseVersionRange(String versionRange) {
```

Comment:
I agree with rob we should just be able to move forward here... there is no harm in breaking this in master.